syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.melon.account";
option java_outer_classname = "AccountProto";
option objc_class_prefix = "ACC";

package account;

service Account {
  rpc requestUserLogin  (LoginRequest)    returns (CodeReply) {}
  rpc requestUserSign   (SignRequest)     returns (CodeReply) {}
  rpc checkConnect      (ConnectRequest)  returns (CodeReply) {}
  rpc requestLogout     (LogoutRequest)   returns (CodeReply) {}
  rpc refreshToken      (RefreshRequest)   returns (CodeReply) {}
}

message LoginRequest {
  string account = 1;
  string password = 2;
}

message SignRequest {
  string account = 1;
  string password = 2;
}

message ConnectRequest {
  string token = 1;
}

message LogoutRequest {
  string token = 1;
}

message RefreshRequest {
  string token = 1;
  string refresh_token = 2;
}

message CodeReply {
  int32 code = 1;
  string msg = 2;
  string data = 3;
}




